src/main/java/org/truffleruby/core/thread/ThreadManager.java

```diff
@@ -481,8 +515,11 @@ public void unregisterThread(DynamicObject thread) {
     }
 
     private void checkCalledInMainThreadRootFiber() {
-        if (getCurrentThread() != rootThread) {
-            throw new UnsupportedOperationException("ThreadManager.shutdown() must be called on the root Ruby Thread");
+        final DynamicObject currentThread = getCurrentThread();
+        if (currentThread != rootThread) {
+            throw new UnsupportedOperationException(
+                    "ThreadManager.shutdown() must be called on the root Ruby Thread (" +
+                            rootThread + ") but was called on " + currentThread);
```

Comment:
This might look a lot better as a format string rather than concatenation.

Reply:
Indeed, will fix.

Reply:
Done.